### PR TITLE
faas-cli: 0.13.13 -> 0.13.15

### PR DESCRIPTION
**Pull request by welteki**
###### Motivation for this change

https://github.com/openfaas/faas-cli/releases/tag/0.13.15

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/faas-cli/default.nix [hunk 1]
```diff
@@ -9,15 +9,15 @@ in
 buildGoModule rec {
   pname = "faas-cli";
   # When updating version change rev.
-  version = "0.13.13";
-  rev = "72816d486cf76c3089b915dfb0b66b85cf096634";
+  version = "0.13.15";
+  rev = "b562392b12a78a11bcff9c6fca5a47146ea2ba0a";
   platform = faasPlatform stdenv.targetPlatform;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/faas-cli/default.nix:14]: Doesn't this belong into meta?

Comment (welteki) [reply pkgs/development/tools/faas-cli/default.nix:14]: We already discussed the need for this [here](https://github.com/NixOS/nixpkgs/pull/137102#discussion_r739041577). If it is considered better practice to put this in meta shall move move it.

Comment (thiagokokada) [reply pkgs/development/tools/faas-cli/default.nix:14]: Sorry, I am not seeing the discussion in the link (it is related to the `rev`, not `platform`).

It is somewhere else :thinking: ?

**Review by thiagokokada — COMMENTED** [review]
Also, this derivation is missing `meta.platforms`.

File: pkgs/development/tools/faas-cli/default.nix (hunk 1 above)
Comment (welteki) [reply pkgs/development/tools/faas-cli/default.nix:14]: You are right, my mistake. I was confused and thought this was about the `rev` again.

The `platform` value is used in the linker flags. The `faasPlatform` function maps the cpu name to the format that is expected by faas-cli.

https://github.com/NixOS/nixpkgs/blob/255cebd8bc1abe11362751f96312c6e63e3e93c8/pkgs/development/tools/faas-cli/default.nix#L3-L7

**Comment by welteki** [comment]
> Also, this derivation is missing `meta.platforms`.

Is it necessary to explicitly set this? When I browse the repo there are a lot of packages that do not have `meta.platforms`.

**Comment by thiagokokada** [comment]
> > Also, this derivation is missing `meta.platforms`.
> 
> Is it necessary to explicitly set this? When I browse the repo there are a lot of packages that do not have `meta.platforms`.

Not necessary, but I think it is good to have `meta.platforms` set.

File: pkgs/development/tools/faas-cli/default.nix (hunk 1 above)
Comment (thiagokokada) [reply pkgs/development/tools/faas-cli/default.nix:14]: Ok, this I understood looking at the code. However do you need to expose it as a attribute? Couldn't you leave it on `let in` instead?

Comment (SuperSandro2000) [reply pkgs/development/tools/faas-cli/default.nix:14]: I think if we inline it in line 33 there is no potential for confusion.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 145080` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>faas-cli</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 145080` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>faas-cli</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]
Result of `nixpkgs-review pr 145080` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>faas-cli</li>
  </ul>
</details>

**Pull request merged**